libs/generic-view/utils/src/lib/data-search/data-search.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { get, isString } from "lodash"
+import { DataProviderSearchConfig } from "device/models"
+import { UseViewFormContext } from "generic-view/utils"
+
+export const dataSearch = (formContext: UseViewFormContext) => {
+  return (
+    data: Record<string, unknown>[] = [],
+    config?: DataProviderSearchConfig
+  ) => {
+    if (!config) {
+      return data
+    }
+    const { fields, phraseSource, minPhraseLength = 0 } = config
+    let searchPhrase = ""
+
+    switch (phraseSource.type) {
+      case "form-fields": {
+        const form = formContext(phraseSource.formKey)
+        searchPhrase = form.watch(phraseSource.field) || ""
+      }
+    }
+
+    if (searchPhrase.length < minPhraseLength) {
+      return undefined
```

Comment:
🟡 

```suggestion
      return []
```

Instead of returning `undefined`, returning an empty array might be worth it. 🤔 

This change promotes type consistency—the function always returns an array, which simplifies further result processing and removes the need to check for undefined in the logic that consumes the function's output. 😊

Reply:
Already discussed